### PR TITLE
Eth JSON-RPC: support passing uint64 in JSON-RPC arguments for EthUint64

### PR DESCRIPTION
**Pull request by ychiaoli18**
## Related Issues
Fixes https://github.com/filecoin-project/ref-fvm/issues/1365

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: chain/types/ethtypes/eth_types.go [hunk 1]
```diff
@@ -40,18 +40,24 @@ func (e EthUint64) MarshalJSON() ([]byte, error) {
...
+		*e = eint
+		return nil
+	} else if eint, err := strconv.ParseUint(string(b), 10, 64); err == nil {
+		*e = EthUint64(eint)
+		return nil
 	}
-	eint := EthUint64(parsedInt)
-	*e = eint
-	return nil
+	return fmt.Errorf("cannot parse %s into EthUint64", string(b))
```
Comment (raulk) [inline chain/types/ethtypes/eth_types.go:60]: This message is externally visible, and users don't understand internal types of Lotus.

```suggestion
	return fmt.Errorf("cannot interpret %s as a hex-encoded uint64, or a number", string(b))
```

**Review by raulk — COMMENTED** [review]
Could you please add a test for `eth_feeHistory` here? We should be testing all styles of arguments.

Plus points if you investigate exactly what MetaMask, ethers.js, web3.js, and other tools use and add test cases for those, since we know this method is a bit of a snowflake.

File: node/impl/full/eth.go [hunk 2]
```diff
@@ -633,19 +637,23 @@ func (a *EthModule) EthFeeHistory(ctx context.Context, blkCount ethtypes.EthUint
...
+	ret := ethtypes.EthFeeHistory{
 		OldestBlock:   ethtypes.EthUint64(oldestBlkHeight),
 		BaseFeePerGas: baseFeeArray,
 		GasUsedRatio:  gasUsedRatioArray,
-	}, nil
+	}
+	if params.RewardPercentiles != nil {
+		reward := make([][]ethtypes.EthBigInt, 0)
+		ret.Reward = &reward
+	}
```
Comment (magik6k) [inline node/impl/full/eth.go:655]: Is this a todo, or is there actually a valid reason why we always populate ret.Reward with an empty array?

**Review by magik6k — APPROVED** [review]
Looks good, just one question + conflicts need resolving

File: node/impl/full/eth.go (hunk 2 above)
Comment (pl-kevin) [reply node/impl/full/eth.go:655]: This is a todo that we should probably implement in the future.

Comment (ychiaoli18) [reply node/impl/full/eth.go:655]: This is a todo that we should probably implement in the future.

Comment (raulk) [reply node/impl/full/eth.go]: So this should return:

> An array of effective priority fee per gas data points from a single block.

We need to calculate the requested percentiles of _effective_ gas premium based on the newest block (I presume it's the newest, need to dig in). _Effective_ means we're clamped at the gas_fee_cap - base_fee.

Comment (raulk) [reply node/impl/full/eth.go]: https://github.com/filecoin-project/lotus/issues/10236

**Review by magik6k — APPROVED** [review]

**Pull request merged**